File: benchmarks/pandas/bench_interval_range_fn.py
"""
Benchmark: pandas.interval_range() — generate equal-length intervals.
Outputs JSON: {"function": "interval_range_fn", "mean_ms": ..., "iterations": ..., "total_ms": ...}
"""
import json
import time
import pandas as pd

WARMUP = 5
ITERATIONS = 100

for _ in range(WARMUP):
    pd.interval_range(start=0, end=100, periods=1000)
    pd.interval_range(start=0, end=1, freq=0.001)
    pd.interval_range(start=0, end=50, periods=500, closed="left")

start = time.perf_counter()
for _ in range(ITERATIONS):
    pd.interval_range(start=0, end=100, periods=1000)
    pd.interval_range(start=0, end=1, freq=0.001)
    pd.interval_range(start=0, end=50, periods=500, closed="left")
total = (time.perf_counter() - start) * 1000

print(json.dumps({
    "function": "interval_range_fn",
    "mean_ms": round(total / ITERATIONS, 3),
    "iterations": ITERATIONS,
    "total_ms": round(total, 3),
}))

File: benchmarks/tsb/bench_interval_range_fn.ts
/**
 * Benchmark: intervalRange — generate a sequence of equal-length intervals.
 * Mirrors pandas.interval_range().
 * Outputs JSON: {"function": "interval_range_fn", "mean_ms": ..., "iterations": ..., "total_ms": ...}
 */
import { intervalRange } from "../../src/index.ts";

const WARMUP = 5;
const ITERATIONS = 100;

for (let i = 0; i < WARMUP; i++) {
  intervalRange(0, 100, { periods: 1000 });
  intervalRange(0, 1, { freq: 0.001 });
  intervalRange(0, 50, { periods: 500, closed: "left" });
}

const start = performance.now();
for (let i = 0; i < ITERATIONS; i++) {
  intervalRange(0, 100, { periods: 1000 });
  intervalRange(0, 1, { freq: 0.001 });
  intervalRange(0, 50, { periods: 500, closed: "left" });
}
const total = performance.now() - start;

console.log(
  JSON.stringify({
    function: "interval_range_fn",
    mean_ms: Math.round((total / ITERATIONS) * 1000) / 1000,
    iterations: ITERATIONS,
    total_ms: Math.round(total * 1000) / 1000,
  }),
);
